File: samples/Julia/associative-array-iteration.jl
mydict = [ "hello"=>13, "world"=>31, "!"=>71 ]

# applying a function to key-value pairs:
 map(println, mydict);

# iterating over key-value pairs:
for (key,value) in mydict
  println("key = $key, value = $value")
end

# iterating over keys:
for key in keys(mydict)
  println("key = $key")
end

# iterating over values:
for value in values(mydict)
  println("value = $value")
end

File: samples/Python/associative-array-iteration.py
myDict = { "hello": 13,
	   "world": 31,
	   "!"    : 71 }

# iterating over key-value pairs:
for key, value in myDict.items():
    print ("key = %s, value = %s" % (key, value))

# iterating over keys:
for key in myDict:
    print ("key = %s" % key)
# (is a shortcut for:)
for key in myDict.keys():
    print ("key = %s" % key)

# iterating over values:
for value in myDict.values():
    print ("value = %s" % value)
